compiler/semexprs.nim

```diff
@@ -873,6 +873,9 @@ proc evalAtCompileTime(c: PContext, n: PNode): PNode =
           localError(c.config, n.info, errCannotInterpretNodeX % renderTree(call))
         else: result = fixupTypeAfterEval(c, result, n)
       else:
+        let noneCtProc = sfCompileTime notin n[0].typ.owner.flags
+        if callee.kind == skConst and (n.typ == nil and noneCtProc):
```

Comment:
Why would `n.typ` be nil?

Reply:
it goes `semConstExpr` -> `fixupTypeAfterEval` -> `semmacrosanity.annotateType` then no branch matches.